walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,113 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> validTimes = new ArrayList<>();  // return of all valid ranges
+    List<TimeRange> invalidTimes = new ArrayList<>();  // keeps track of invalid ranges
+    Collection<String> attendees = request.getAttendees();
+
+    int duration = (int) request.getDuration();
+
+    // if length of meeting is longer than a whole day, return empty list
+    if (duration > TimeRange.WHOLE_DAY.duration()) {
+        return validTimes;
+    }
+
+    // if no conflicting events
+    if (events.isEmpty()) {
+        validTimes.add(TimeRange.WHOLE_DAY);
+        return validTimes;
+    }
+
+    // Add to invalid times collection
+    for (Event event : events) {
+        for(String attendee : attendees) {
```

Comment:
To make this more efficient, you can use a hashset for attendees, something like:
```
Set<String> attendees = new HashSet<>(request.getAttendees());
for (Event event : events) {
  for (String attendee : event.getAttendees()) {
    if (attendees.contains(attendee)) {
      invalidTimes.add(event.getWhen());
    }
  }
}
```